-1 
Tested at: b25910acb2189eb7a4c08f7522e3d592d2a56f6c
 I found an error when building: 
 
 <pre>In file included from /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-10-1100/src/DataFormats/PatCandidates/interface/Lepton.h:22:0,
                 from /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-10-1100/src/DataFormats/PatCandidates/interface/Muon.h:27,
                 from /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-10-1100/src/PhysicsTools/PatAlgos/test/private/PATUserDataTestModule.cc:35:
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-10-1100/src/DataFormats/PatCandidates/interface/PATObject.h: In instantiation of 'void pat::PATObject<ObjectType>::addUserData(const string&, const T&, bool, bool) [with T = ROOT::Math::LorentzVector<ROOT::Math::PxPyPzE4D<double> >; ObjectType = reco::Muon; std::string = std::basic_string<char>]':
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-10-1100/src/PhysicsTools/PatAlgos/test/private/PATUserDataTestModule.cc:129:60:   required from here
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-10-1100/src/DataFormats/PatCandidates/interface/PATObject.h:310:91: error: no matching function for call to 'pat::PATObject<reco::Muon>::addUserDataObject_(const string&, std::auto_ptr<pat::UserData>, bool&)'
         addUserDataObject_( label, pat::UserData::make<T>(data, transientOnly), overwrite );
                                                                                           ^
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-10-1100/src/DataFormats/PatCandidates/interface/PATObject.h:310:91: note: candidate is:
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-10-1100/src/DataFormats/PatCandidates/interface/PATObject.h:771:8: note: void pat::PATObject<ObjectType>::addUserDataObject_(const string&, pat::UserData*, bool) [with ObjectType = reco::Muon; std::string = std::basic_string<char>]
   void PATObject<ObjectType>::addUserDataObject_( const std::string & label, pat::UserData * data, bool overwrite ) 
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12750/10254/summary.html

--- a/DataFormats/Common/interface/OwnVector.h
+++ b/DataFormats/Common/interface/OwnVector.h
@@ -46,7 +46,6 @@ namespace edm {
       typedef T const& reference;
       typedef ptrdiff_t difference_type;
       typedef typename base::const_iterator::iterator_category iterator_category;
-      const_iterator(typename base::const_iterator const& it) : i(it) { }
       const_iterator(iterator const& it) : i(it.i) { }
       const_iterator() {}
       const_iterator& operator++() { ++i; return *this; }
@@ -65,9 +64,11 @@ namespace edm {
       const_iterator & operator +=(difference_type d) { i += d; return *this; }
       const_iterator & operator -=(difference_type d) { i -= d; return *this; }
       reference operator[](difference_type d) const { return *const_iterator(i+d); } // for boost::iterator_range []
-      typename base::const_iterator base_iter() const { return i; }
     private:
+      const_iterator(typename base::const_iterator const& it) : i(it) { }
+      typename base::const_iterator base_iter() const { return i; }
       typename base::const_iterator i;
+      friend class OwnVector<T,P>;
     };
     class iterator {
     public:
@@ -76,7 +77,6 @@ namespace edm {
       typedef T & reference;
       typedef ptrdiff_t difference_type;
       typedef typename base::iterator::iterator_category iterator_category;
-      iterator(typename base::iterator const& it) : i(it) { }
       iterator() {}
       iterator& operator++() { ++i; return *this; }
       iterator operator++(int) { iterator ci = *this; ++i; return ci; }
@@ -96,6 +96,7 @@ namespace edm {
       iterator & operator -=(difference_type d) { i -= d; return *this; }
       reference operator[](difference_type d) const { return *iterator(i+d); } // for boost::iterator_range []
     private:
+      iterator(typename base::iterator const& it) : i(it) { }
       typename base::iterator i;
       friend class const_iterator;
       friend class OwnVector<T, P>;

--- a/DataFormats/PatCandidates/interface/PATObject.h
+++ b/DataFormats/PatCandidates/interface/PATObject.h
@@ -307,13 +307,15 @@ namespace pat {
       /// unless transientOnly is set to true
       template<typename T>
       void addUserData( const std::string & label, const T & data, bool transientOnly=false, bool overwrite=false ) {
-        addUserDataObject_( label, pat::UserData::make<T>(data, transientOnly), overwrite );
+        std::auto_ptr<pat::UserData> made(pat::UserData::make<T>(data, transientOnly));
+        addUserDataObject_( label, made, overwrite );
       }
 
       /// Set user-defined data. To be used only to fill from ValueMap<Ptr<UserData>>
       /// Do not use unless you know what you are doing.
       void addUserDataFromPtr( const std::string & label, const edm::Ptr<pat::UserData> & data, bool overwrite=false ) {
-        addUserDataObject_( label, data->clone(), overwrite );
+        std::auto_ptr<pat::UserData> cloned(data->clone());
+        addUserDataObject_( label, cloned, overwrite );
       }
 
       /// Get user-defined float
@@ -455,7 +457,7 @@ namespace pat {
       /// if (kinResolutions_.size() == kinResolutionLabels_.size()+1), then the first resolution has no label.
       std::vector<std::string>            kinResolutionLabels_;
 
-      void addUserDataObject_( const std::string & label, pat::UserData * value, bool overwrite = false ) ;
+      void addUserDataObject_( const std::string & label, std::auto_ptr<pat::UserData> & value, bool overwrite = false ) ;
 
     private:
       const pat::UserData *  userDataObject_(const std::string &key) const ;
@@ -768,7 +770,7 @@ namespace pat {
   }
 
   template <class ObjectType>
-  void PATObject<ObjectType>::addUserDataObject_( const std::string & label, pat::UserData * data, bool overwrite ) 
+  void PATObject<ObjectType>::addUserDataObject_( const std::string & label, std::auto_ptr<pat::UserData> & data, bool overwrite ) 
   {
     auto it = std::lower_bound(userDataLabels_.begin(), userDataLabels_.end(), label);
     const auto dist = std::distance(userDataLabels_.begin(), it);

